src/Primary_Extension/org/lobobrowser/primary/ext/ComponentSource.java

```diff
@@ -693,11 +704,15 @@ public void navigateOrSearch() {
   public void search() {
     final ToolsSettings settings = ToolsSettings.getInstance();
     final SearchEngine searchEngine = settings.getSelectedSearchEngine();
+    final String addressText = this.addressField.getText();
     if (searchEngine != null) {
       try {
-        final String addressText = this.addressField.getText();
-        assert (addressText.charAt(0) == '?');
-        this.navigate(searchEngine.getURL(addressText.substring(1)));
+        if (addressText.charAt(0) == '?') {
+          assert (addressText.charAt(0) == '?');
+          this.navigate(searchEngine.getURL(addressText.substring(1)));
+        } else {
+        this.navigate(searchEngine.getURL(addressText));
+        }
```

Comment:
This change prevents the Search Button on the browser from dropping the first character in the string; since the other way to call the search function is by starting the query with a question mark.